### PR TITLE
Wf 439 e attribute table

### PR DESCRIPTION
**Pull request by boehlerlukas**
## Pull request
The c-attributegird takes an array of objects, with the properties name, content & url, and renders them into a key/value like grid. On mobile only the first entry will be displayed. There is a headline option through the boolean headline prop, which underlines the first entry if set.

INFO: I accidentally implemented the mobile ticket within the first run too, so both tickets should be reviewed within this PR. Hope that's ok.

### Ticket
https://winterhalter-fenner.atlassian.net/projects/WF/issues/WF-486
https://winterhalter-fenner.atlassian.net/projects/WF/issues/WF-573
 
### Browser testing
- http://localhost:6060/styleguide#cattributegrid
- http://localhost:8080/styleguide/attributegrid
 
### Checklist
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Did run automated tests and linters
- [x] Did assign ticket
- [x] Double checked target branch
 
## Review/Test checklist
- [x] Did run automated tests and linters
- [x] Did review code
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Re-assign ticket to developer

### CODE REVIEW

File: app/components/c-attributegrid.vue [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+          <span :class="b('name')">{{ attribute.name }}</span>
+        </div>
+        <div :class="b('col')">
+          <a v-if="attribute.url" :class="b('link')" :href="attribute.url">{{ attribute.content }}</a>
+          <span v-else :class="b('content')">{{ attribute.content }}</span>
+        </div>
+      </div>
+    </div>
+    <div :class="b('toggle', { toggled: toggled })" @click="toggle()">
+      <img :class="b('arrow')" src="../assets/icons/arrowDown.svg">
```
[the remark this answers is not in the document]
Comment (patric-eberle) [reply app/components/c-attributegrid.vue]: Yes i would also recommend the `i-` prefix. `i-<name>--<variant>.svg`

File: app/components/c-attributegrid.vue [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      },
+
+      /**
+       * Defines weather the first row is a headline or not.
+       */
+      headline: {
+        type: Boolean,
+        default: false,
+      },
+    },
```
[the remark this answers is not in the document]
Comment (boehlerlukas) [reply app/components/c-attributegrid.vue]: Yes! Totally correct. Haven't thought about that.

File: app/components/c-attributegrid.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (boehlerlukas) [reply app/components/c-attributegrid.vue]: :+1: perfect

File: app/components/c-attributegrid.vue [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    &__link {
+      color: $color-grayscale--400;
+      padding: $spacing--0;
+      border-color: $color-grayscale--400;
+    }
+
+    &--headline &__row:first-child &__col {
+      font-weight: $font-weight--bold;
```
[the remark this answers is not in the document]
Comment (boehlerlukas) [reply app/components/c-attributegrid.vue]: It's ok the way it's implemented. #939393 is wrong :)

File: app/components/c-attributegrid.vue [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    &__arrow {
+      display: block;
+      height: $spacing--15;
+      width: auto;
+      margin: auto;
+      transition: transform $toggle-animation-duration;
+    }
+
+    &__toggle--open &__arrow {
+      transform: rotate(180deg) !important;
```
Comment (patric-eberle) [inline app/components/c-attributegrid.vue]: - [ ] Is `!important` really needed? Bad practice. Removing the `important` also worked for me. If needed better try to increase selector weight.

File: app/components/c-attributegrid.vue [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    &__table {
+      transition: none;
+      max-height: inherit;
+      overflow: visible;
+    }
+    
+    &--dont-shrink-on-mobile &__table {
+      max-height: inherit !important;
```
Comment (patric-eberle) [inline app/components/c-attributegrid.vue]: - [ ] Is `!important` really needed? Bad practice. You could also define the inline `max-height` value dynamically based on the open/close (`tableMaxHeight`/`auto`) state, so you probably won't even have to overwrite it in CSS.

File: app/components/c-attributegrid.vue [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+      @include media(xs) {
+        transition: none;
+        max-height: inherit;
+        overflow: visible;
+      }
+    }
+
+    &__table--open {
+      @include media(xs) {
+        max-height: inherit !important;
```
Comment (patric-eberle) [inline app/components/c-attributegrid.vue]: - [ ] `!important` really needed?

File: app/components/c-attributegrid.vue [hunk 7]
```diff
@@ -0,0 +1,220 @@
+<template>
+  <div :class="b({ headline: headline, shrinkOnMobile: shrinkOnMobile, dontShrinkOnMobile: !shrinkOnMobile })">
```
Comment (patric-eberle) [inline app/components/c-attributegrid.vue]: - [ ] When defining more than 2 attributes please move to computed.

**Comment by patric-eberle** [comment]
@boehlerlukas there are also linting issues. Please run `npm run test` and fix.

**Comment by patric-eberle** [comment]
@boehlerlukas "Shrink on mobile" demo in styleguide is also broken (has text overflow)

**Comment by boehlerlukas** [comment]
@patric-eberle i tried several ways, but can't to get it to work properly (without too much JS) without one !important, which toggles the max-height for desktops. Pushing right now, maybe you can think of one way, which won't involve subscribing to resize events o.a.

File: app/components/c-attributegrid.vue [hunk 8]
```diff
@@ -187,33 +218,29 @@
...
-    }
 
     &--shrink-on-mobile &__table {
       transition: max-height $toggle-animation-duration;
       overflow: hidden;
       
       @include media(xs) {
         transition: none;
-        max-height: inherit;
+        max-height: inherit !important;
```
Comment (patric-eberle) [inline app/components/c-attributegrid.vue]: - [ ] Throws a Stylelint error (`npm run test`). Please add `// stylelint-disable-line declaration-no-important`

File: app/components/c-attributegrid.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+### Default
```
Comment (patric-eberle) [inline app/components/c-attributegrid.md]: - [ ] Rename file according to component.

File: app/components/c-attributegrid.vue [hunk 10]
```diff
@@ -0,0 +1,247 @@
+<template>
```
Comment (patric-eberle) [inline app/components/c-attributegrid.vue]: - [ ] Rename file according to component.

File: app/components/c-attributegrid.vue [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+  <div :class="b(modifiers)">
+    <div :class="b('table', { open: isOpen })" :style="{ 'max-height': tableMaxHeight }">
+      <div
+        v-for="(attribute, index) in attributes"
```
Comment (patric-eberle) [inline app/components/c-attributegrid.vue]: - [ ] Please update template according to new attribute data structure from backend:

```
"tech_attributes" : [
	{
		"key": "someKey",
		"value": "someValue",
		"url": "URL for search || empty"
	},
	{}
]
```

File: app/styleguide/routes/attributegrid.vue [hunk 12]
```diff
@@ -0,0 +1,29 @@
+<template>
+  <div :class="b('spacing')">
+    <h1>Attributegrid</h1>
+    <c-attribute-grid :attributes="[{'name': 'Test', 'content': 'Testvalue'}, {'name': 'Test', 'content': 'Testvalue'}, {'name': 'Test', 'content': 'Testvalue'}, {'name': 'Test', 'content': 'Testvalue'}, {'name': 'Test', 'content': 'Testvalue', url: 'http://www.google.com'}]" />
```
Comment (patric-eberle) [inline app/styleguide/routes/attributegrid.vue]: - [ ] Please also add to http://localhost:8080/styleguide/product-detail page (`specs`)

File: app/components/c-attributegrid.vue (hunk 11 above)
Comment (patric-eberle) [reply app/components/c-attributegrid.vue]: Note: discussion with BE if we should also include an emphasis flag is still open. (so we don't have to guess which item we should make bold)

**Pull request closed without merging**